mempool/clist_mempool.go

```diff
@@ -539,6 +504,19 @@ func (mem *CListMempool) notifyTxsAvailable() {
 	}
 }
 
+func (mem *CListMempool) TxsRemoved() <-chan types.TxKey {
+	return mem.txsRemoved
+}
+
+func (mem *CListMempool) notifyTxRemoved(tx types.TxKey) {
+	if mem.txsRemoved != nil {
+		select {
+		case mem.txsRemoved <- tx:
+		default:
```

Comment:
Should log that the tx was not removed as it could be considered a leak

Reply:
Who should log that a tx was not removed? In `CListMempool`, the function `RemoveTxByKey` returns an error when it cannot remove a transaction. The only case when this happens is when the transaction key was not found in `txsMap`. Then it returns an error, which is properly logged by caller of `RemoveTxByKey`. 

Should we also notify the mempool reactor when `RemoveTxByKey` could not remove a transaction? What I will do is  to call `mem.notifyTxRemoved(txKey)` in `RemoveTxByKey` regardless of whether `txKey` is in `txsMap`. Then the mempool reactor will update the senders for `txKey` in any case.

Reply:
I mean that if the message is not placed in `txsRemoved` it will not be processed in `updateSendersRoutine` and the entry for txId will not be removed from the senders map.